### DISCUSSION TITLE
Is there any build hook?

### DISCUSSION THREAD

**Discussion by jeertmans** [PyO3/maturin#1936] [Q&A] [answered]
Hello, I am currently wondering if it is possible to hook some script during the build process?

The idea is that I need to modify the `README.md` file that is put inside the `sdist` folder. Is it possible, like with PDM's build backend, to hook on some build step (like `pre_build`)?

**Comment by messense** [comment, marked as answer]
No build hooks at the moment.

**Reply by jeertmans** [reply]
Thanks for the reply! Is it something you would consider in the future?

**Comment by DavidVujic** [comment]
Having hooks in a similar way as with Hatch/hatchling (or PDM) would be a very nice addition. I develop tooling that is on top of Dependency & package managment tools like Hatch, PDM, Poetry & uv and find that very useful. The tooling collects relative includes and adds it to the build process as local includes, before the wheels are built.

**Comment by net** [comment]
I'd also like this to generate stub files on build.
